### ISSUE TITLE
Custom 'cancel' button calls Cancel(id) method, but upload process won't skip to next file

### ISSUE THREAD

**Issue by notken** [FineUploader/fine-uploader#1481]
This is in reference to http://stackoverflow.com/questions/32913006/fine-uploader-when-cancelling-an-in-progress-file-any-other-queued-files-remai

I've changed our implementation from what was listed at SO, and we're now using qq.FineUploader with a template. However for various design reasons, we tried to implement our own Cancel button. We removed the template one, and instead, in a javascript click event, called

	var id = $container.attr("qq-file-id");
	uploader.cancel(id);

instead. This worked fine in terms of cancelling in-progress files, or files awaiting uploading. But if we cancelled an in-progress upload, any other files awaiting upload sit in the queue, but the queue won't be processed. In fact the only way to get awaiting files uploaded is to add a new file to the queue. That new file gets uploaded first, and then any awaiting files get uploaded until the queue is empty.

We were only testing locally, so I should point out that although we use chunking and multiple uploads, the cancelled in-progress file was a 150MB video to give us time to cancel. So all available chunks and calls are being made on behalf of the one cancelled file.

**Comment by rnicholus** [comment]
As in the SO question you referenced, I am unable to reproduce this issue. Were this a Fine Uploader issue, I would have expected other reports by now, as this is a fairly significant one. The code in your SO question was a bit hard to follow and overly complex, so I suggested you start over and gradually re-introduce that complexity (when necessary) until you are able to reproduce the issue. Once you are at that point and are able to provide a concise set of code and steps to reproduce, I'd be happy to re-open and look further.

**Comment by notken** [comment]
Yeah, I said on SO and in the first comment above that we're not using the code you saw any more. I'm using the UI with templating. I can repeat it with pretty much vanilla code, adding in only my own click handler. I've left Fine Uploader's button in the following template, which works fine. But if I click my own Cancel, which calls .cancel(id) on the the uploader, it doesn't restart the queue.



	    <script src="/Scripts/shared/all.fine-uploader/all.fine-uploader.min.js"></script>
		<link href="/Scripts/shared/all.fine-uploader/fine-uploader-new.min.css" rel="stylesheet" />

	    <script type="text/template" id="qq-template-manual-trigger">
        <div class="qq-uploader-selector qq-uploader" qq-drop-area-text="Drop files here">
            <div class="qq-total-progress-bar-container-selector qq-total-progress-bar-container">
                <div role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" class="qq-total-progress-bar-selector qq-progress-bar qq-total-progress-bar"></div>
            </div>
            <div class="qq-upload-drop-area-selector qq-upload-drop-area" qq-hide-dropzone>
                <span class="qq-upload-drop-area-text-selector"></span>
            </div>
            <div class="buttons">
                <div class="qq-upload-button-selector qq-upload-button">
                    <div>Select files</div>
                </div>
                <button type="button" id="trigger-upload" class="btn btn-primary">
                    <i class="icon-upload icon-white"></i> Upload
                </button>
            </div>
            <span class="qq-drop-processing-selector qq-drop-processing">
                <span>Processing dropped files...</span>
                <span class="qq-drop-processing-spinner-selector qq-drop-processing-spinner"></span>
            </span>
            <ul class="qq-upload-list-selector qq-upload-list" aria-live="polite" aria-relevant="additions removals">
                <li>
                    <div class="qq-progress-bar-container-selector">
                        <div role="progressbar" aria-valuenow="0" aria-valuemin="0" aria-valuemax="100" class="qq-progress-bar-selector qq-progress-bar"></div>
                    </div>
                    <span class="qq-upload-spinner-selector qq-upload-spinner"></span>
                    <img class="qq-thumbnail-selector" qq-max-size="100" qq-server-scale>
                    <span class="qq-upload-file-selector qq-upload-file"></span>
                    <span class="qq-edit-filename-icon-selector qq-edit-filename-icon" aria-label="Edit filename"></span>
                    <input class="qq-edit-filename-selector qq-edit-filename" tabindex="0" type="text">
                    <span class="qq-upload-size-selector qq-upload-size"></span>
                    <button type="button" class="qq-btn qq-upload-cancel-selector qq-upload-cancel">Your cancel</button>
                    <button type="button" class="qq-btn customcancel">My cancel</button>
                    <span role="status" class="qq-upload-status-text-selector qq-upload-status-text"></span>
                </li>
            </ul>

        </div>
    </script>

	    <style>
    		#trigger-upload {
    			color: white;
    			background-color: #00ABC7;
    			font-size: 14px;
    			padding: 7px 20px;
    			background-image: none;
    		}

    		#fine-uploader-manual-trigger .qq-upload-button {
    			margin-right: 15px;
    		}

    		#fine-uploader-manual-trigger .buttons {
    			width: 36%;
    		}

    		#fine-uploader-manual-trigger .qq-uploader .qq-total-progress-bar-container {
    			width: 60%;
    		}
    	</style>

    
    <div id="fine-uploader-manual-trigger"></div>

    
	    <script>
    		var manualUploader = new qq.FineUploader({
    			element: document.getElementById('fine-uploader-manual-trigger'),
    			template: 'qq-template-manual-trigger',
    			request: {
    				endpoint: "server/upload"
    			},
    			thumbnails: {
    				placeholders: {
    					waitingPath: '/Scripts/shared/all.fine-uploader/placeholders/waiting-generic.png',
    					notAvailablePath: '/Scripts/shared/all.fine-uploader/placeholders/not_available-generic.png'
    				},
    				maxCount: 50
    			},
    			debug: true,
    			maxConnections: 10,
    			retry: {
    				enableAuto: true,
    				showButton: true
    			},
    			chunking: {
    				partSize: 1048576,
    				enabled: true,
    				mandatory: true,
    				concurrent: {
    					enabled: true
    				},
    				success: {
    					endpoint: "server/success"
    				}
    			},
    			resume: {
    				enabled: true
    			}
    		});

	    	$("body").on("click", ".customcancel", function () {
			    var $li = $(this).closest("li");
	    		var id = $li.attr("qq-file-id");
	    		manualUploader.cancel(id);
			    $li.fadeOut();
		    });
    	</script>

This is what I'm testing. If I have a large file as file id 0, and a second file queued up to go, and then cancel file 0 while it's in progress, file 1 never starts. If I drag a new image into the drag box, file 2 will upload, and when it's finished, file 1 will start uploading.

**Comment by rnicholus** [comment]
The problem is in your code. [The `cancel` method](http://docs.fineuploader.com/branch/master/api/methods.html#cancel), as the documentation states, requires an `Number`/Integer parameter to be passed in as the file ID. You are passing a string. You can have avoided this by making use of Fine Uploader's API to get the file ID, instead of sniffing DOM attributes.

For example, your click handler could be rewritten as:

```javascript
$("body").on("click", ".customcancel", function () {
   var $li = $(this).closest("li"),
         id = manualUploader.getId($li[0]);
   manualUploader.cancel(id);
   $li.fadeOut();
});
```

**Comment by notken** [comment]
OK, I can confirm this does fix it. The weird thing is, though, that it does actually cancel. It just doesn't carry on.

**Comment by rnicholus** [comment]
yes, this is not the first time someone has tried to pass a string as an ID and experienced strange results.
